### ISSUE TITLE
Cider-quit not working correctly

### ISSUE THREAD

**Issue by malchmih** [clojure-emacs/cider#1472]
I'm getting `nrepl-send-sync-request: Wrong type argument: stringp, nil` now, when calling `cider-quit`. Versions that I use:
```clojure
; CIDER 0.11.0snapshot (package: 20151214.1327) (Java 1.8.0_66, Clojure 1.7.0, nREPL 0.2.12)
v.r.core> 
```

**Comment by expez** [comment]
Hi

Could you `M-x toggle-debug-on-error <RET>`, reproduce the problem, and post the resulting stacktrace?

Which OS is this one?  Have you been tracking the snapshot, or did this happen after an upgrade across versions?

**Comment by malchmih** [comment]
```
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  nrepl-send-request(("op" "close" "session" "6672b191-5ccf-446d-9f90-563527e2637a") #[257 "\301\300\"\207" [(dict) nrepl--merge] 4 "\n\n(fn RESP)"] nil)
  nrepl-send-sync-request(("op" "close" "session" "6672b191-5ccf-446d-9f90-563527e2637a") nil)
  nrepl-sync-request:close(nil "6672b191-5ccf-446d-9f90-563527e2637a")
  cider--close-buffer(#<buffer *cider-repl localhost*>)
  cider--close-connection-buffer(#<buffer *cider-repl localhost*>)
  cider--quit-connection(#<buffer *cider-repl localhost*>)
  cider-quit(nil)
  call-interactively(cider-quit nil nil)
  command-execute(cider-quit)
```
I'm using Mac OS, been using snapshot the whole time.

**Comment by bbatsov** [comment]
I can confirm this. Damn, we keep breaking this all the time...

**Comment by arrdem** [comment]
I'm impacted by this one as well.

**Comment by hugoduncan** [comment]
This same error also occurs when trying to close a repl from the connection browser.

**Comment by cap10morgan** [comment]
I am unable to reproduce this in my local v0.10 branch where I've backported all the fixes on master since 0.10.0 was released. So hopefully it's fixed there.

**Comment by bbatsov** [comment]
I can no longer reproduce this on the current `master`. Guess we fixed it without even noticing.
